butane_core/src/codegen/migration.rs

```diff
@@ -46,6 +46,10 @@ fn create_atables(ast_struct: &ItemStruct, config: &dbobj::Config) -> Vec<ATable
             .clone()
             .expect("db object fields must be named")
             .to_string();
+        if name == *"pub_time" {
+            // Temporary solution to skip pub_time field in butane tests
+            continue;
+        }
```

Comment:
This is causing the CI failure when running with the `datetime` feature enabled.

As this macro is being run with the cfg's not evaluated and thus fields removed, the .table files contain all of the fields, causing a mismatch with the ORM objects.

I think we need to extract the cfgs, and evaluate them explicitly.

Reply:
It looks like there isnt any built in Rust support for this. c.f. https://github.com/rust-lang/rust/issues/90765

Maybe a crate which can help...

Reply:
https://github.com/EmbarkStudios/cfg-expr looks like it can help.

Reply:
`CARGO_CFG_FEATURE` is only available in `build.rs` , so this approach would require that there is a build.rs, and it saves `CARGO_CFG_FEATURE` somewhere so that the migration code can know which features will be active during compilation.

Reply:
An alternative is to add a "cfg attributes" member to `AColumn` , so that the .table files describe the fact that some columns are conditional based on features.

This sounds more sensible, as then the .table file doesnt change based on which features are enabled, but the migration .sql still needs to be different depending on the features enabled.  That problem can be avoided if the migrations are generated at runtime (https://github.com/Electron100/butane/issues/313) or the migrations are written in Rust which can also use feature gates (https://github.com/Electron100/butane/issues/315).

The other approach to .sql files is that separate copies would be needed for each combination of features that are referred to in the schema definition.